low-level/files/opt/seagate/sspl/setup/sspl_config.py

```diff
@@ -48,9 +48,38 @@ class SSPLConfig:
     def __init__(self):
         """Init method for sspl setup config."""
         self._script_dir = os.path.dirname(os.path.abspath(__file__))
-        self.role = None
-        self.config_index = 'sspl'
-        Conf.load(self.config_index, consts.sspl_config_path)
+        # Load sspl and global configs
+        Conf.load(consts.SSPL_CONFIG_INDEX, consts.sspl_config_path)
+        global_config_url = Conf.get(
+            consts.SSPL_CONFIG_INDEX, "SYSTEM_INFORMATION>global_config_copy_url")
+        Conf.load(consts.GLOBAL_CONFIG_INDEX, global_config_url)
+
+    def validate(self):
+        """Validate config for supported role and product"""
+        # Validate role
+        self.role = Conf.get(consts.GLOBAL_CONFIG_INDEX, 'release>setup')
+        if not self.role:
+            raise SetupError(
+                errno.EINVAL,
+                "%s - validation failure. %s",
+                self.name,
+                "Role not found in global config copy.")
+        if self.role not in consts.setups:
+            raise SetupError(
+                errno.EINVAL,
+                "%s - validation failure. %s",
+                self.name,
+                "Role '%s' is not supported. Check Usage" % self.role)
+
+        # Validate Product
+        self.product = Conf.get(consts.GLOBAL_CONFIG_INDEX, 'release>product')
```

Comment:
> release>product

already getting validated in sspl_postinstall, why we need again here? I think we can skip here.

Reply:
Yes, product validation is already happening in sspl_post_install.py.
Now, product validation is removed from sspl_config.py

Reply:
Done.